ejviola/mailroom.py

```diff
@@ -0,0 +1,104 @@
+#!/usr/bin/env python3
+
+# Create a list to hold lists of donations
+donordata = []
+
+# Start list with some sample donations
+donordata.append(["Bill Gates", 3000000])
+donordata.append(["Bill Gates", 1125000])
+donordata.append(["Bill Gates",100125])
+donordata.append(["Elon Musk", 1500000])
+donordata.append(["Paul Allen", 2000000])
+donordata.append(["Paul Allen", 134000])
+donordata.append(["Mark Zuckerberg", 125000])
+donordata.append(["Jeff Bezos", 1])
+
+# While loop to let user send thank you notes or view a report
+while True:
+    choice = float(input("Choose one of the following (Enter 1, 2, or 3): \n 1) Send a Thank You \n 2) Create a Report \n 3) Quit\n"))
+
+    # Create a list of unique donors
+    list = []
+    for i in donordata:
+        # row = str(donordata[i]).split(",").strip("[").strip("]")
+        row = i[0]
+        # If that donor is already in the list, do nothing
+        if row in list:
+            counter = []
+        # Otherwise, add that donor
+        else:
+            list.append(row)
+    if choice == 1:
+        print("You chose: send a thank you\n")
+        while True:
+            # Ask the user to enter a name or type 'list'
+            entername = str(input("Please enter the full name of the donor you would like to thank, \nenter 'list' to see a list of donors, \nor enter 'back' to go back to the main directory:\n"))
+            # Construct the list of donors
+
+            # If the user asks for a list of donors, print the list
+            if entername.lower() == "list":
+                print(list)
+                print("\n")
+                break
+            elif entername.lower() == "back":
+                break
+            else:
+                if entername in list:
+                    print("You selected "+entername+"\n")
+                    try:
+                        amount = float(input("How much did "+entername+" donate?\n"))
+                    except:
+                        print("Error: not a valid amount \n")
+                    donordata.append([entername, amount])
+
+                else:
+                    #add to list
+                    list += entername
```

Comment:
This creates an item for each character in the name that was entered.  When I entered the name `Todd` is produced:
`['Bill Gates', 'Elon Musk', 'Paul Allen', 'Mark Zuckerberg', 'Jeff Bezos', 'T', 'o', 'd', 'd']`